tests/acceptance/SubscribeOnRegistrationPageCest.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace MailPoet\Test\Acceptance;
+
+class SubscribeOnRegistrationPageCest {
+  function allowSubscribeOnRegistrationPage(\AcceptanceTester $I) {
+    $I->wantTo('Allow users to subscribe to lists on site registration page');
+    $I->login();
+    //Go to settings
+    $I->amOnMailPoetPage('Settings');
+    $I->seeInCurrentUrl('page=mailpoet-settings');
+    $I->checkOption('#settings[subscribe_on_register]');
+    //work around funky legacy code with brittle xpath selectors
+    $I->click(['xpath'=>'//*[@id="mailpoet_subscribe_in_form"]/p[3]/span/span[1]/span/ul/li/input']);
+    $I->waitForText('My First List', 5);
+    $I->fillField(['xpath'=>'//*[@id="mailpoet_subscribe_in_form"]/p[3]/span/span[1]/span/ul/li/input'], 'My First List');
+    $I->pressKey(['xpath'=>'//*[@id="mailpoet_subscribe_in_form"]/p[3]/span/span[1]/span/ul/li/input'], \WebDriverKeys::ENTER);
```

Comment:
And why didn't you use the `selectOptionInSelect2` helper function here again?

Reply:
The settings page, much like the import/export page under Subscribers, isn't quite as well structured as the rest of the codebase - using xpath here was the least-restrictive working selector. I always start with the cleanest available option (data-ID) and move down the list from there until I find a method that works consistently.

Reply:
that is not what I was asking here. I was asking about helper function `selectOptionInSelect2` instead of using `fillField` and `pressKey`